filebeat/harvester/log.go

```diff
@@ -99,6 +99,13 @@ func (h *Harvester) Harvest() {
 			event.Text = &text
 			event.JSONFields = message.Fields
 			event.EventMetadata = h.config.EventMetadata
+			if message.Fields != nil {
+				if event.EventMetadata.Fields != nil {
+					event.EventMetadata.Fields.Update(message.Fields)
```

Comment:
This will get more obvious when is merged: https://github.com/elastic/beats/pull/2290 Meta data addition and merging should not be a task of the harvester.